photon-api/src/main/scala/com/linkedin/photon/ml/function/svm/SmoothedHingeLossFunction.scala

```diff
@@ -94,19 +94,22 @@ object SmoothedHingeLossFunction {
    * @return A function which builds the appropriate type of [[ObjectiveFunction]] for a given [[Coordinate]] type and
    *         optimization settings.
    */
-  def buildFactory(treeAggregateDepth: Int): (CoordinateOptimizationConfiguration) => ObjectiveFunction =
-    (config: CoordinateOptimizationConfiguration) => {
-      config match {
-        case fEOptConfig: FixedEffectOptimizationConfiguration =>
+  def buildFactory
+      (treeAggregateDepth: Int)
+      (config: CoordinateOptimizationConfiguration): Option[Int] => ObjectiveFunction =
+
+    config match {
+      case fEOptConfig: FixedEffectOptimizationConfiguration =>
+        (interceptIndexOpt: Option[Int]) =>
```

Comment:
To clarify, will interceptIndexOpt be used in DistributedSmoothedHingeLossFunction

Reply:
No - I did some tests and found that the smoothed-hinge loss function works significantly worse than logistic regression. In a future PR, I plan to get rid of it. I told Bill in advance, so that he wouldn't waste his time updating the smoothed-hinge classes.